pkg/restore/restore.go

```diff
@@ -1677,15 +1678,55 @@ func (ctx *restoreContext) restoreItem(obj *unstructured.Unstructured, groupReso
 		errs.Add(namespace, err)
 		return warnings, errs, itemExists
 	}
+
 	if patchBytes != nil {
-		if _, err = resourceClient.Patch(name, patchBytes); err != nil {
-			ctx.log.Errorf("error patch for managed fields %s: %v", kube.NamespaceAndName(obj), err)
-			if !apierrors.IsNotFound(err) {
-				errs.Add(namespace, err)
-				return warnings, errs, itemExists
+		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			// First attempt to patch the object
+			if patchBytes != nil {
+				_, err = resourceClient.Patch(name, patchBytes)
 			}
-		} else {
-			ctx.log.Infof("the managed fields for %s is patched", kube.NamespaceAndName(obj))
+			if patchBytes == nil || err != nil {
+				ctx.log.Errorf("error patching managed fields %s: %v", kube.NamespaceAndName(obj), err)
```

Comment:
The above error log is appropriate if we just called `Patch` above, but not necessarily if `patchBytes` is nil. We could put another `if` block around it right here, or we could move it to after line 1686 above with that if block.